### PR TITLE
Issue 1282 - feature that generates AWS CodeBuild files

### LINKED ISSUE #1282: Create a feature which generates AWS CodeBuild files
see: https://docs.aws.amazon.com/codebuild/index.html
see: https://github.com/micronaut-projects/micronaut-starter/issues/1280

### PR DESCRIPTION
**Pull request by wetted**
closes #1282

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    runtime-versions:
+      java: corretto@jdkVersion.majorVersion()
+  pre_build:
+    commands:
+      - echo Nothing to do in the pre_build phase...
+  build:
+    commands:
+      - echo Build started on `date`
+@if (buildTool.equals(BuildTool.MAVEN)) {
+      - mvn install
```
Comment (alvarosanchez) [inline starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw]: The equivalent of `gradle build` is `mvn verify`. `mvn install` would be comparable to `gradle publishToMavenLocal`, which for the scope of a CI job I think it's not necessary.

Also, shouldn't this be using the wrapper?

```suggestion
      - mvn verify
```

**Review by alvarosanchez — CHANGES_REQUESTED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw (hunk 1 above)
Comment (wetted) [reply starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw]: @alvarosanchez I'll make the change to use the `verify` target as you suggested. I should probably make sure it's consistent for the Github and GCP CI workflow features too.

I tried using the maven wrapper and I get errors from AWS CodeBuild. If there is a way to make that work in the `buildspec.yml` file I will make the change.

```
[INFO] 
[INFO] -------------------------< awscimvn:awscimvn >--------------------------
[INFO] Building awscimvn 0.1
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.813 s
[INFO] Finished at: 2022-06-23T21:43:27Z
[INFO] ------------------------------------------------------------------------
[ERROR] Unknown lifecycle phase "/root/.m2". You must specify a valid lifecycle phase or a goal in the format <plugin-prefix>:<goal> or <plugin-group-id>:<plugin-artifact-id>[:<plugin-version>]:<goal>. Available lifecycle phases are: validate, initialize, generate-sources, process-sources, generate-resources, process-resources, compile, process-classes, generate-test-sources, process-test-sources, generate-test-resources, process-test-resources, test-compile, process-test-classes, test, prepare-package, package, pre-integration-test, integration-test, post-integration-test, verify, install, deploy, pre-clean, clean, post-clean, pre-site, site, post-site, site-deploy. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/LifecyclePhaseNotFoundException

[Container] 2022/06/23 21:43:27 Command did not exit successfully ./mvnw verify exit status 1
[Container] 2022/06/23 21:43:27 Phase complete: BUILD State: FAILED
[Container] 2022/06/23 21:43:27 Phase context status code: COMMAND_EXECUTION_ERROR Message: Error while executing command: ./mvnw verify. Reason: exit status 1
[Container] 2022/06/23 21:43:27 Entering phase POST_BUILD
[Container] 2022/06/23 21:43:27 Running command echo Build completed on `date`
Build completed on Thu Jun 23 21:43:28 UTC 2022
```

Comment (alvarosanchez) [reply starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw]: The wrapper error is weird. Looks like if it were executing `mvn /root/.m2`.

If `mvn` comes preinstalled with CodeBuild, then I think it's fine to use it.

Comment (wetted) [reply starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw]: I agree. It’s a weird error. Using `mvn` builds fine, and that’s what the example `buildspec.yml` I based on uses. Thanks for the feedback.

Comment (alvarosanchez) [reply starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw]: Updated the suggestion snippet to use `mvn verify`. Apply it at your convenience.

**Review by alvarosanchez — APPROVED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/AWSCiWorkflowFeature.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Override
+    public String getTitle() {
+        return "AWS CodeBuild CI Workflow";
+    }
+
+    @Override
+    @NonNull
+    public String getDescription() {
+        return "Adds a AWS CodeBuild Workflow to build a Micronaut application";
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/AWSCiWorkflowFeature.java]: replace `Workflow` with `build specification`. I don't think CodeBuild refers to its build configuration as workflow.  https://docs.aws.amazon.com/codebuild/latest/userguide/concepts.html

File: starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    runtime-versions:
+      java: corretto@jdkVersion.majorVersion()
+  pre_build:
+    commands:
+      - echo Nothing to do in the pre_build phase...
+  build:
+    commands:
+      - echo Build started on `date`
+@if (buildTool.equals(BuildTool.MAVEN)) {
+      - mvn verify
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw:23]: Use Maven wrapper

File: starter-core/src/test/groovy/io/micronaut/starter/feature/ci/workflows/aws/AWSWorkflowCISpec.groovy [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    void 'test gcp-workflow-ci wrapper validation and upload is created for Maven'() {
+        when:
+        def output = generate(ApplicationType.DEFAULT,
+                new Options(Language.JAVA, TestFramework.JUNIT, BuildTool.MAVEN, JdkVersion.DEFAULT_OPTION),
+                [AWSCiWorkflowFeature.NAME])
+        def workflow = output["buildspec.yml"]
+
+        then:
+        workflow
+        workflow.contains("- mvn")
```
Comment (sdelamo) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/ci/workflows/aws/AWSWorkflowCISpec.groovy:38]: change `mvn` for wrapper.

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw (hunk 3 above)
Comment (alvarosanchez) [reply starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw:23]: Why? Maven appears to be pre-installed in CodeBuild. Using the wrapper will make the build to download Maven on every run

File: starter-core/src/test/groovy/io/micronaut/starter/feature/ci/workflows/aws/AWSWorkflowCISpec.groovy (hunk 4 above)
Comment (alvarosanchez) [reply starter-core/src/test/groovy/io/micronaut/starter/feature/ci/workflows/aws/AWSWorkflowCISpec.groovy:38]: See https://github.com/micronaut-projects/micronaut-starter/pull/1301#discussion_r905487170

File: starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw (hunk 3 above)
Comment (wetted) [reply starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw:23]: If I use the Maven wrapper I get an odd error, whereas the non-wrapper version works fine: see [#1301 (comment)](https://github.com/micronaut-projects/micronaut-starter/pull/1301#discussion_r905498784)

Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/aws/templates/buildSpec.rocker.raw:23]: Leave `mvn`

**Pull request merged**